dockstore-webservice/src/main/java/io/dockstore/webservice/resources/MetadataResource.java

```diff
@@ -414,4 +418,23 @@ public Config getConfig() {
         }
     }
 
+    @GET
+    @Timed
+    @UnitOfWork(readOnly = true)
+    @Path("cli-version")
+    @Operation(summary = "Get Dockstore CLI latest version.", description = "Get Dockstore CLI latest version. NO authentication")
+    @ApiOperation(value = "Get Dockstore CLI latest version.", notes = "Get Dockstore CLI latest version. NO authentication")
+    public String getCliVersion() {
+        GHRelease ghRelease;
+        try {
+            GitHub github = new GitHubBuilder().build();
+            GHRepository repository = github.getRepository("dockstore/dockstore-cli");
+            ghRelease = repository.getLatestRelease();
```

Comment:
Two thoughts:

1. You may run into GitHub rate limiting issues; not sure if our cache handles anonymous requests. Trying invoking your endpoint 1000 times with `ab` and see. I'd think you want to cache the response even if you don't run into those issues.
2. Since the CLI can upgrade to the latest unstable, I'd return the latest unstable release as well (the goal is for the CLI to use this endpoint as well, isn't it?), and it could give us the flexibility to surface it in the UI if we choose.

Reply:
@coverbeck Is there an example of how to cache the response?
When would the CLI use this endpoint?

Reply:
>     1. Is there an example of how to cache the response?

We use a Caffeine cache in at least [one place](https://github.com/dockstore/dockstore/blob/db15223af216d10633e6fbd08000080d8b60ffde/dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/CacheConfigManager.java#L91). Might be overkill for one value, but may be just fine.

>     2. When would the CLI use this endpoint?

The CLI figures out the latest stable and unstable version and (optionally) upgrades to either. Presumably it's making GitHub calls directly? It could just call this endpoint instead. It doesn't have to change if it's working, but maybe at some point.

Reply:
Did not use cache, but did now capture unstable version

Reply:
@denis-yuen @coverbeck Is the guidance to implement caching or not since a Java object is now returned?

Reply:
> @denis-yuen @coverbeck Is the guidance to implement caching or not since a Java object is now returned?

The point of the cache is:

1. To avoid running into GitHub rate limits
2. To make the endpoint snappy.

Caching is still needed.

Reply:
I added caching, but not sure if I did it correctly.